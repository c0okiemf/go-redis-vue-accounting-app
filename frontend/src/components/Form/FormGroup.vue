<script setup lang="ts">
defineProps<{
  label: string;
  error?: string;
}>();
</script>

<template>
  <div class="form-group">
    <label for="accountNumber">{{ label }}</label>
    <slot></slot>
    <span v-if="error" class="error">{{ error }}</span>
  </div>
</template>

<style lang="scss" scoped>
.form-group {
  width: 100%;
  display: flex;
  flex-direction: column;
  position: relative;
}

label {
  font-size: 0.8rem;
}

.error {
  color: red;
  font-size: 0.8rem;
  position: absolute;
  bottom: -1.2rem;
}
</style>
