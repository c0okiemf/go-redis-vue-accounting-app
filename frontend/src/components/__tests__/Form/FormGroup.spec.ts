import { describe, it, expect } from 'vitest';

import { mount } from '@vue/test-utils';
import FormGroup from '../../Form/FormGroup.vue';

describe('FormGroup', () => {
  it('renders the label properly', () => {
    const wrapper = mount(FormGroup, { props: { label: 'Hello Vitest' } });
    expect(wrapper.text()).toContain('Hello Vitest');
  });

  it('renders an error if supplied', () => {
    const wrapper = mount(FormGroup, { props: { label: 'Hello Vitest', error: 'I am error' } });
    expect(wrapper.text()).toContain('I am error');
  });

  it('matches snapshot', () => {
    const wrapper = mount(FormGroup, { props: { label: 'Hello Vitest' } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders a label element', () => {
    const wrapper = mount(FormGroup, { props: { label: 'Hello Vitest' } });
    expect(wrapper.find('label').exists()).toBe(true);
  });

  it('renders an error message element if error is supplied', () => {
    const wrapper = mount(FormGroup, { props: { label: 'Hello Vitest', error: 'I am error' } });
    expect(wrapper.find('.error').exists()).toBe(true);
  });

  it('handles missing error prop gracefully', () => {
    const wrapper = mount(FormGroup, { props: { label: 'Hello Vitest' } });
    expect(wrapper.find('.error-message').exists()).toBe(false);
  });

  it('handles empty label prop', () => {
    const wrapper = mount(FormGroup, { props: { label: '' } });
    expect(wrapper.find('label').text()).toBe('');
  });
});
