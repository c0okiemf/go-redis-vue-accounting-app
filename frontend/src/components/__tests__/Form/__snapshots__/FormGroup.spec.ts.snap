// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormGroup > matches snapshot 1`] = `
"<div data-v-59a90264="" class="form-group"><label data-v-59a90264="" for="accountNumber">Hello Vitest</label>
  <!--v-if-->
</div>"
`;
